threeML/plugins/OGIPLike.py

```diff
@@ -1,1420 +1,72 @@
-        # than the size of the monte carlo interval.
-
-        def integral(e1, e2):
-            # Simpson's rule
-
-            return (e2 - e1) / 6.0 * (differential_flux(e1)
-                                      + 4 * differential_flux((e1 + e2) / 2.0)
-                                      + differential_flux(e2))
-
-        return differential_flux, integral
-
-    def use_effective_area_correction(self, min_value=0.8, max_value=1.2):
-        """
-        Activate the use of the effective area correction, which is a multiplicative factor in front of the model which
-        might be used to mitigate the effect of intercalibration mismatch between different instruments.
-
-        NOTE: do not use this is you are using only one detector, as the multiplicative constant will be completely
-        degenerate with the normalization of the model.
+        assert is_valid_variable_name(name), "Name %s is not a valid name for a plugin. You must use a name which is " \
+                                             "a valid python identifier: no spaces, no operators (+,-,/,*), " \
+                                             "it cannot start with a number, no special characters" % name
 
-        NOTE2: always keep at least one multiplicative constant fixed to one (its default value), when using this
-        with other OGIPLike-type detectors
+        # Read the pha file (or the PHAContainer instance)
 
-        :param min_value: minimum allowed value (default: 0.8, corresponding to a 20% - effect)
-        :param max_value: maximum allowed value (default: 1.2, corresponding to a 20% + effect
-        :return:
-        """
+        pha = PHASpectrum(observation,spectrum_number=spectrum_number,file_type='observed',rsp_file=response,arf_file=arf_file)
 
-        self._nuisance_parameter.free = True
-        self._nuisance_parameter.bounds = (min_value, max_value)
 
-        # Use a uniform prior by default
 
-        self._nuisance_parameter.set_uninformative_prior(Uniform_prior)
+        # Get the required background file, response and (if present) arf_file either from the
+        # calling sequence or the file.
+        # NOTE: if one of the file is specified in the calling sequence, it will be used whether or not there is an
+        # equivalent specification in the header. This allows the user to override the content of the header of the
+        # PHA file, if needed
 
-    def fix_effective_area_correction(self, value=1):
-        """
-        Fix the multiplicative factor (see use_effective_area_correction) to the provided value (default: 1)
+        if background is None:
 
-        :param value: new value (default: 1, i.e., no correction)
-        :return:
-        """
+            background = pha.background_file
 
-        self._nuisance_parameter.value = value
-        self._nuisance_parameter.fix = True
+            assert background is not None, "No background file provided, and the PHA file does not specify one."
```

Comment:
:) I was curious about that this morning and saw this just now. I can add this right now.